src/latch_cli/centromere/ast_parsing.py

```diff
@@ -0,0 +1,126 @@
+        # custom
+        "custom_task",
+        "custom_memory_optimized_task",
+        # nf
+        "nextflow_runtime_task",
+        # l40s gpu
+        "g6e_xlarge_task",
+        "g6e_2xlarge_task",
+        "g6e_4xlarge_task",
+        "g6e_8xlarge_task",
+        "g6e_12xlarge_task",
+        "g6e_16xlarge_task",
+        "g6e_24xlarge_task",
+        # v100 gpu
+        "v100_x1_task",
+        "v100_x4_task",
+        "v100_x8_task",
+    }
+
+
+class Visitor(ast.NodeVisitor):
+    def __init__(self, file: Path):
+        self.file = file
+        self.flyte_objects: list[FlyteObject] = []
+
+    def visit_FunctionDef(self, node: ast.FunctionDef):
+        if len(node.decorator_list) == 0:
+            return self.generic_visit(node)
+
+        for decorator in node.decorator_list:
+            if isinstance(decorator, ast.Name):
+                if decorator.id == "workflow":
+                    self.flyte_objects.append(FlyteObject("workflow", node.name))
+                elif is_task_decorator(decorator.id):
+                    self.flyte_objects.append(FlyteObject("task", node.name))
+
+            elif isinstance(decorator, ast.Call):
+                func = decorator.func
+                assert isinstance(func, ast.Name)
+
+                if not is_task_decorator(func.id) and func.id != "workflow":
+                    continue
+
+                if func.id == "workflow":
+                    self.flyte_objects.append(FlyteObject("workflow", node.name))
+                    continue
+
+                dockerfile: Optional[Path] = None
+                for kw in decorator.keywords:
+                    if kw.arg != "dockerfile":
+                        continue
+
+                    try:
+                        dockerfile = Path(ast.literal_eval(kw.value)).resolve()
+                    except ValueError as e:
+                        click.secho(
+                            dedent(f"""\
+                                There was an issue parsing the `dockerfile` argument for task `{node.name}` in {self.file}.
+                                Note that values passed to `dockerfile` must be string literals.
+                            """),
+                            fg="red",
+                        )
+
+                        raise click.exceptions.Exit(1) from e
+
+                    if not dockerfile.exists():
+                        click.secho(
+                            f"The `dockerfile` value ({dockerfile}) for task `{node.name}` in {self.file} does not exist.",
+                            fg="red",
+                        )
+
+                        raise click.exceptions.Exit(1)
+
+                self.flyte_objects.append(FlyteObject("task", node.name, dockerfile))
+
+        return self.generic_visit(node)
+
+
+def get_flyte_objects(file: Path) -> list[FlyteObject]:
+    res = []
+    if file.is_dir():
+        for child in file.iterdir():
+            res.extend(get_flyte_objects(child))
```

Comment:
might want to use a queue instead of recursing
the file tree might be very deep for some stupid reason

also might want to avoid recursing into directories that don't have a `__init__.py` since those cannot contain importable python files anyway